### ISSUE TITLE
Walkontable Table methods return wrong information in the overlay

### ISSUE THREAD

**Issue by warpech** [handsontable/handsontable#6191]
### Description
In Walkontable `table.js` class, the methods such as:

```
getFirstRenderedRow
getFirstVisibleRow
getFirstRenderedColumn
getFirstVisibleColumn
getLastRenderedRow
getLastVisibleRow
getLastRenderedColumn
getLastVisibleColumn
isRowBeforeRenderedRows
isRowAfterViewport
isColumnBeforeViewport
isColumnBeforeRenderedColumns
isColumnAfterViewport
isColumnAfterRenderedColumns
```

return incorrect information when used in overlays. This is because they are based on the information from `wtViewport`, which contains *global* calculations, which might or might not be true for an overlay.

Some methods contain workarounds, which you can see by searching for `Overlay.isOverlayTypeOf`, but not all of them.

As a rule of thumb, I think that every method which uses `wtViewport`, should also have an if for `Overlay.isOverlayTypeOf`.

Perhaps, for better perofrmance, it might make sense to extend the `Table` class for every kind of overlay and overload these helper methods. It that case, all checks for  `Overlay.isOverlayTypeOf` could be completely avoided.

I am not sure if there are any actual bugs in Handsontable due to this issue, but I assume there might be.

### Steps to reproduce

1. See the demo. The console shows the expected and actual values
2.
3.

### Demo

https://jsfiddle.net/frjk2873/

### Your environment
* Handsontable version: 7.1.0
* Browser Name and version: Chrome
* Operating System: Windows

**Comment by warpech** [comment]
Partial fix for this problem is in PR: https://github.com/handsontable/handsontable/pull/6157/files#diff-0a25aaaa72d09482c0875ad092179097 (look at `table.js` changes between the lines 640 and 856).

**Comment by warpech** [comment]
This was fixed in the PR https://github.com/handsontable/handsontable/pull/6200

**Comment by aninde** [comment]
This bug will be fixed in `7.2.0` version.  An error no longer exists in console and function returns correct data.

**Demo**
`7.2.0` v. https://jsfiddle.net/aninde/stv8rdx1/
